mlx/ops.cpp

```diff
@@ -753,6 +753,43 @@ array repeat(const array& arr, int repeats, StreamOrDevice s) {
   return repeat(flatten(arr, s), repeats, 0, s);
 }
 
+array tile(
+    const array& arr,
+    std::vector<int> reps,
+    StreamOrDevice s /* = {} */) {
+  auto shape = arr.shape();
+  if (reps.size() < shape.size()) {
+    reps.insert(reps.begin(), shape.size() - reps.size(), 1);
+  }
+  if (reps.size() > shape.size()) {
+    shape.insert(shape.begin(), reps.size() - shape.size(), 1);
+  }
+
+  std::vector<int> expand_shape;
+  std::vector<int> broad_shape;
+  std::vector<int> final_shape;
+  int odims = reps.size() - shape.size();
+  for (int i = 0; i < shape.size(); i++) {
+    if (reps[i] != 1) {
+      expand_shape.push_back(1);
+      broad_shape.push_back(reps[i]);
+    }
+    expand_shape.push_back(shape[i]);
+    broad_shape.push_back(shape[i]);
+
+    final_shape.push_back(reps[i]);
+    if (odims > 0) {
+      odims -= 1;
+    } else {
+      final_shape.back() *= shape[i];
+    }
```

Comment:
Since `reps` and `shape` are now normalized to have the same size. The above can be removed and changed with a simple one-liner `final_shape.push_back(reps[i] * shape[i])`.